src/libraries/System.Private.CoreLib/src/System/Buffers/SharedArrayPool.cs

```diff
@@ -523,7 +523,7 @@ private static bool TryGetInt32EnvironmentVariable(string variable, out int resu
         {
             // Avoid globalization stack, as it might in turn be using ArrayPool.
 
-            if (Environment.GetEnvironmentVariableCore_NoArrayPool(variable) is string envVar &&
+            if (Environment.GetEnvironmentVariableCore_NoArrayPool(variable) is { } envVar &&
                 envVar.Length is > 0 and <= 32) // arbitrary limit that allows for some spaces around the maximum length of a non-negative Int32 (10 digits)
```

Comment:
```suggestion
            string? envVar = Environment.GetEnvironmentVariableCore_NoArrayPool(variable);
            if (envVar is { Length: > 0 and <= 32 }) // arbitrary limit that allows for some spaces around the maximum length of a non-negative Int32 (10 digits)
      
```

@jkotas what is your opinion on this pattern?

Reply:
I think it is about as good as the pattern in main. Some people prefer multiline statements, some people prefer single line statements. (Personally, I tend to prefer single line statements when everything else is equal.)